federation/merge_schemas.go

```diff
@@ -312,6 +338,36 @@ func mergeEnumValues(a, b []introspectionEnumValue, mode MergeMode) ([]introspec
 	return merged, nil
 }
 
+func mergeDirectives(a, b []introspectiveDirective, mode MergeMode) ([]introspectiveDirective, error) {
+	directives := make(map[string][]introspectiveDirective)
+	for _, a := range a {
+		directives[a.Name] = append(directives[a.Name], a)
+	}
+	for _, b := range b {
+		directives[b.Name] = append(directives[b.Name], b)
+	}
+	names := make([]string, 0, len(directives))
+	for name := range directives {
+		names = append(names, name)
+	}
+	sort.Strings(names)
+
+	merged := make([]introspectiveDirective, 0, len(names))
+	for _, name := range names {
+		p := directives[name]
+		if len(p) == 1 {
```

Comment:
noob question:

what if the mode is union and the length of p is more than 1, but the two values are different? Then wouldn't we just be taking the first value? Should we check that they are the same similar to how we merge types below?

Reply:
Yeah, if two directives had the same names, we would only take the first directive. I believe this is the same behaviour as how we merge type fields as well. However, as Zhekai pointed out, the directives of an introspection query are actually hardcoded, so I will be removing this method altogether.